File: src/hasoftware-library/src/hasoftware/api/DeviceType.java
package hasoftware.api;

public enum DeviceType {

    UNKNNOWN(0, "UNKNOWN"),
    NASCOCG(260, "NASCOCG"), //     NASCO CAN Gatway
    POINT(200, "POINT"), //         Point
    NASCOCAIM(201, "NASCOCAIM"), // NASCO CAN Analog Input Module
    NASCOCOM(202, "NASCOCOM"), //   NASCO CAN Output Module
    KIRKDECT(261, "KIRKDECT"), //   Kirk DECT phone
    EMAIL(263, "EMAIL"), //         Email Address
    ANDROID(265, "ANDROID"), //     Andriod Modile Phone (via Notify My Android)
    SMS(262, "SMS"), //             Mobile Phone (via SMS Web Service)
    SMS2(268, "SMS2"), //           Mobile Phone (via F1103)
    PAGER(264, "PAGER"), //         Alphanumeric Pager
    TEMP(266, "TEMP"), //           Temperature Sensor
    SENSOR(267, "SENSOR"); //       Sensor

    private final int _id;
    private final String _code;

    DeviceType(int id, String code) {
        _id = id;
        _code = code;
    }

    public int getId() {
        return _id;
    }

    public String getCode() {
        return _code;
    }

    private static DeviceType[] allDeviceTypes;

    public static DeviceType fromId(int id) {
        if (allDeviceTypes == null) {
            allDeviceTypes = DeviceType.values();
        }
        for (int i = 0; i < allDeviceTypes.length; i++) {
            if (allDeviceTypes[i]._id == id) {
                return allDeviceTypes[i];
            }
        }
        return DeviceType.UNKNNOWN;
    }
}

File: src.net/hasoftware-library/Api/DeviceType.cs
using System.Collections.Generic;

namespace hasoftware.Api
{
    public class DeviceType
    {
        public static readonly DeviceType UNKNNOWN = new DeviceType(0, "UNKNOWN");
        public static readonly DeviceType NASCOCG = new DeviceType(260, "NASCOCG");     // NASCO CAN Gatway
        public static readonly DeviceType POINT = new DeviceType(200, "POINT");         // Point
        public static readonly DeviceType NASCOCAIM = new DeviceType(201, "NASCOCAIM"); // NASCO CAN Analog Input Module
        public static readonly DeviceType NASCOCOM = new DeviceType(202, "NASCOCOM");   // NASCO CAN Output Module
        public static readonly DeviceType KIRKDECT = new DeviceType(261, "KIRKDECT");   // Kirk DECT phone
        public static readonly DeviceType EMAIL = new DeviceType(263, "EMAIL");         // Email Address
        public static readonly DeviceType ANDROID = new DeviceType(265, "ANDROID");     // Andriod Modile Phone (via Notify My Android)
        public static readonly DeviceType SMS = new DeviceType(262, "SMS");             // Mobile Phone (via SMS Message)
        public static readonly DeviceType PAGER = new DeviceType(264, "PAGER");         // Alphanumeric Pager
        public static readonly DeviceType TEMP = new DeviceType(266, "TEMP");           // Temperature Sensor
        public static readonly DeviceType SENSOR = new DeviceType(267, "SENSOR");       // Sensor

        private static readonly List<DeviceType> AllDeviceTypes = new List<DeviceType>();

        public int Id { get; private set; }
        public string Code { get; private set; }

        private DeviceType(int id, string code)
        {
            Id = id;
            Code = Code;
            AllDeviceTypes.Add(this);
        }

        public static DeviceType FromId(int id)
        {
            foreach (var deviceType in AllDeviceTypes)
            {
                if (deviceType.Id == id) return deviceType;
            }
            return UNKNNOWN;
        }
    }
}
